spec/gilded_rose_spec.js

```diff
@@ -1,7 +1,166 @@
-describe("Gilded Rose", function() {
+"use strict"
+
+describe("Give Gilded Rose ", function() {
+
+  // Setup:All items have a sell_in value which denotes the number of days we have to sell the item
+  // Setup:All items have a quality value which denotes how valuable the item is
+  // * Assertion Done and refactored Normal Case: At the end of each day our system lowers both values for every item
+  //
+  // Pretty simple, right? Well this is where it gets interesting:
+  // * Assertion done and refactored :Normal Case: Once the sell_in days is less then zero, quality degrades twice as fast
+  // * Aessertion done and refactored: The quality of an item is never negative
+  // * Assertion done and (refactored?) "Aged Brie" actually increases in quality the older it gets
+  // * Assertion done and Tested The quality of an item is never more than 50
+  // * Assertion done and Tested Sulfuras", being a legendary item, never has to be sold nor does it decrease in quality
+  // * Assertion done and tested "Backstage passes", like aged brie, increases in quality as it's sell_in value decreases; quality increases by
+  //    2 when there are 10 days or less and by 3 when there are 5 days or less but quality drops to 0 after the concert
+  // We have recently signed a supplier of conjured items. This requires an update to our system:
+  //    "Conjured" items degrade in quality twice as fast as normal items
+
+
+  // Normal Item
+
+  items=[]
+
+  it("Normal Item: should decrement both sell in and quality after running update_quality", function() {
+    var currSize=items.length;
```

Comment:
Set the `item` array to contain a single item. Either way, you are modifying the contents of the array. Prefer the simplest / smallest data set possible. Imagine someone later adds 5000 items to the items array in the production file. Now your tests take forever to run. 

It is best practice to have test data entirely encapsulated in the tests and entirely isolated from production data.

Using this approach also lets you remove the `currSize` when adds no  value to the functionality under test.